diff --git a/llvm/lib/Target/DADAO/DADAORegisterInfo.h b/llvm/lib/Target/DADAO/DADAORegisterInfo.h
new file mode 100644
index 000000000..02e91eb91
--- /dev/null
+++ b/llvm/lib/Target/DADAO/DADAORegisterInfo.h
@@ -0,0 +1,39 @@
+//===-- DADAORegisterInfo.h - DADAO Register Information -------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLVM_LIB_TARGET_DADAO_DADAOREGISTERINFO_H
+#define LLVM_LIB_TARGET_DADAO_DADAOREGISTERINFO_H
+
+#include "llvm/CodeGen/TargetRegisterInfo.h"
+
+#define GET_REGINFO_HEADER
+#include "DADAOGenRegisterInfo.inc"
+
+namespace llvm {
+
+struct DADAOTargetMachine;
+
+class DADAORegisterInfo : public DADAOGenRegisterInfo {
+public:
+  DADAORegisterInfo();
+
+  const uint16_t *
+  getCalleeSavedRegs(const MachineFunction *MF) const override;
+
+  BitVector getReservedRegs(const MachineFunction &MF) const override;
+
+  bool eliminateFrameIndex(MachineBasicBlock::iterator MI, int SPAdj,
+                           unsigned FIOperandNum,
+                           RegScavenger *RS) const override;
+
+  Register getFrameRegister(const MachineFunction &MF) const override;
+};
+
+} // namespace llvm
+
+#endif // LLVM_LIB_TARGET_DADAO_DADAOREGISTERINFO_H
