diff --git a/target/dadao/cpu-qom.h b/target/dadao/cpu-qom.h
new file mode 100644
index 0000000..099965c
--- /dev/null
+++ b/target/dadao/cpu-qom.h
@@ -0,0 +1,31 @@
+/*
+ * DADAO CPU QOM header (target agnostic)
+ *
+ * Copyright (c) 2026 DADAO Contributors
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms and conditions of the GNU General Public License,
+ * version 2 or later, as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef TARGET_DADAO_CPU_QOM_H
+#define TARGET_DADAO_CPU_QOM_H
+
+#include "hw/core/cpu.h"
+
+#define TYPE_DADAO_CPU "dadao-cpu"
+
+OBJECT_DECLARE_CPU_TYPE(DadaoCPU, DadaoCPUClass, DADAO_CPU)
+
+#define DADAO_CPU_TYPE_SUFFIX "-" TYPE_DADAO_CPU
+#define DADAO_CPU_TYPE_NAME(name) (name DADAO_CPU_TYPE_SUFFIX)
+
+#endif /* TARGET_DADAO_CPU_QOM_H */
